feature(@desktop/chat) Enhance message context menu with mark as unread

kounkou opened the pull request:
### Description

This is the Golang repo PR matching PR : 12879

Feature to enhance message context menu with mark as unread.

[Figma design](https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?type=design&node-id=14632-460085&mode=design&t=9iVc38RXY5QVm5uz-0)

fixes #10329

### What does the PR do

The PR does the following : 

- Adds capacity to mark a message as unread
- Adds capacity to mark a message with mention as unread
- Adds persistence to the marking of the message (change can be seen at after reboot)
- Adds marking in right click contextual menu

### Affected areas

- Chat screen
- Contextual menu (Right click 

### Screenshot of functionality (including design for comparison)

![output](https://github.com/status-im/status-desktop/assets/2589171/390fedbf-cd1b-4969-aff0-55eae9cfbc28)

ghost commented:
Hey @kounkou, and thank you so much for making your first pull request in status-go! :heart: Please help us make your experience better by filling out this brief questionnaire https://goo.gl/forms/uWqNcVpVz7OIopXg2

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (83)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | e607abfe | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/1/) | 2023-11-27 09:43:02 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/1//consoleText) |
| :heavy_check_mark: | e607abfe | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/1/) | 2023-11-27 09:44:20 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231127-093959-e607ab-pr4382.zip) |
| :heavy_check_mark: | e607abfe | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/1/) | 2023-11-27 09:44:58 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231127-093959-e607ab-pr4382.zip) |
| :heavy_check_mark: | e607abfe | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/1/) | 2023-11-27 09:46:05 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231127-093959-e607ab-pr4382.aar) |
| | | | | | | |
| :heavy_multiplication_x: | 9b45e4bf | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/2/) | 2023-11-28 03:20:33 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/2//consoleText) |
| :heavy_check_mark: | 9b45e4bf | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/2/) | 2023-11-28 03:20:57 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231128-031750-9b45e4-pr4382.zip) |
| :heavy_check_mark: | 9b45e4bf | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/2/) | 2023-11-28 03:21:29 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231128-031750-9b45e4-pr4382.zip) |
| :heavy_check_mark: | 9b45e4bf | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/2/) | 2023-11-28 03:23:24 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231128-031747-9b45e4-pr4382.aar) |
| | | | | | | |
| :heavy_check_mark: | 34f3f567 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/3/) | 2023-11-28 07:10:00 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231128-070753-34f3f5-pr4382.aar) |
| :heavy_check_mark: | 34f3f567 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/3/) | 2023-11-28 07:11:11 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231128-070753-34f3f5-pr4382.zip) |
| :heavy_check_mark: | 34f3f567 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/3/) | 2023-11-28 07:11:29 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231128-070756-34f3f5-pr4382.zip) |
| :heavy_multiplication_x: | 34f3f567 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/3/) | 2023-11-28 07:24:58 | ~17 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/3//consoleText) |
| | | | | | | |
| :x: | fff5ec03 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/4/) | 2023-11-28 07:24:59 | ~20 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/4/consoleText) |
| :x: | fff5ec03 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/4/) | 2023-11-28 07:25:04 | ~22 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/4/consoleText) |
| :x: | fff5ec03 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/4/) | 2023-11-28 07:25:05 | ~26 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/4/consoleText) |
| :heavy_multiplication_x: | fff5ec03 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/4/) | 2023-11-28 07:26:08 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/4//consoleText) |
| :heavy_multiplication_x: | fff5ec03 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/5/) | 2023-11-28 07:31:30 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b773eabb | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/5/) | 2023-11-28 07:58:25 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231128-075513-b773ea-pr4382.zip) |
| :heavy_check_mark: | b773eabb | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/5/) | 2023-11-28 08:01:09 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231128-075513-b773ea-pr4382.aar) |
| :heavy_check_mark: | b773eabb | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/5/) | 2023-11-28 08:04:35 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231128-075517-b773ea-pr4382.zip) |
| :heavy_check_mark: | b773eabb | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/6/) | 2023-11-28 08:32:32 | ~37 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b6cf552a | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/6/) | 2023-11-28 08:49:48 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231128-084812-b6cf55-pr4382.zip) |
| :heavy_check_mark: | b6cf552a | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/6/) | 2023-11-28 08:50:16 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231128-084808-b6cf55-pr4382.aar) |
| :heavy_check_mark: | b6cf552a | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/6/) | 2023-11-28 08:51:22 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231128-084812-b6cf55-pr4382.zip) |
| :heavy_multiplication_x: | b6cf552a | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/7/) | 2023-11-28 08:52:00 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/7//consoleText) |
| :heavy_multiplication_x: | b6cf552a | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/8/) | 2023-11-28 09:10:20 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/8//consoleText) |
| :heavy_multiplication_x: | b6cf552a | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/9/) | 2023-11-28 17:37:12 | ~11 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/9//consoleText) |
| :heavy_multiplication_x: | b6cf552a | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/10/) | 2023-11-28 18:07:45 | ~9 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/10//consoleText) |
| :heavy_check_mark: | b6cf552a | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/11/) | 2023-11-28 19:18:15 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/11//consoleText) |
| | | | | | | |
| :heavy_check_mark: | d22292cf | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/7/) | 2023-12-05 08:10:36 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231205-080728-d22292-pr4382.zip) |
| :heavy_check_mark: | d22292cf | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/7/) | 2023-12-05 08:11:54 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231205-080728-d22292-pr4382.aar) |
| :heavy_check_mark: | d22292cf | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/7/) | 2023-12-05 08:12:23 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231205-080727-d22292-pr4382.zip) |
| :heavy_multiplication_x: | d22292cf | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/12/) | 2023-12-05 08:26:09 | ~18 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/12//consoleText) |
| :heavy_multiplication_x: | d22292cf | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/13/) | 2023-12-05 09:00:16 | ~6 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/13//consoleText) |
| :heavy_check_mark: | d22292cf | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/14/) | 2023-12-05 10:04:14 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/14//consoleText) |
| | | | | | | |
| :heavy_check_mark: | c6fbb363 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/8/) | 2023-12-05 21:53:57 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231205-215231-c6fbb3-pr4382.zip) |
| :heavy_check_mark: | c6fbb363 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/8/) | 2023-12-05 21:55:16 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231205-215231-c6fbb3-pr4382.zip) |
| :heavy_check_mark: | c6fbb363 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/8/) | 2023-12-05 21:56:43 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231205-215231-c6fbb3-pr4382.aar) |
| :heavy_multiplication_x: | c6fbb363 | [#15](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/15/) | 2023-12-05 22:18:40 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/15//consoleText) |
| :heavy_multiplication_x: | c6fbb363 | [#16](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/16/) | 2023-12-05 23:53:22 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/16//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 36d5534f | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/9/) | 2023-12-06 04:30:35 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231206-042830-36d553-pr4382.aar) |
| :heavy_check_mark: | 36d5534f | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/9/) | 2023-12-06 04:31:30 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231206-042830-36d553-pr4382.zip) |
| :heavy_check_mark: | 36d5534f | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/9/) | 2023-12-06 04:31:42 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231206-042830-36d553-pr4382.zip) |
| :heavy_multiplication_x: | 36d5534f | [#18](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/18/) | 2023-12-06 04:45:26 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/18//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | b4bc9d58 | [#19](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/19/) | 2023-12-06 06:32:53 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/19//consoleText) |
| :heavy_check_mark: | b4bc9d58 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/10/) | 2023-12-06 06:33:30 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231206-063145-b4bc9d-pr4382.zip) |
| :heavy_check_mark: | b4bc9d58 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/10/) | 2023-12-06 06:33:50 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231206-063141-b4bc9d-pr4382.aar) |
| :heavy_check_mark: | b4bc9d58 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/10/) | 2023-12-06 06:34:55 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231206-063141-b4bc9d-pr4382.zip) |
| | | | | | | |
| :heavy_check_mark: | d506b592 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/11/) | 2023-12-06 07:52:42 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231206-075117-d506b5-pr4382.zip) |
| :heavy_check_mark: | d506b592 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/11/) | 2023-12-06 07:52:59 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231206-075117-d506b5-pr4382.aar) |
| :heavy_check_mark: | d506b592 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/11/) | 2023-12-06 07:54:59 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231206-075117-d506b5-pr4382.zip) |
| :heavy_check_mark: | d506b592 | [#20](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/20/) | 2023-12-06 08:24:18 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/20//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 4aa7f108 | [#21](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/21/) | 2023-12-06 18:45:39 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/21//consoleText) |
| :heavy_check_mark: | 4aa7f108 | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/12/) | 2023-12-06 18:46:05 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231206-184428-4aa7f1-pr4382.zip) |
| :heavy_check_mark: | 4aa7f108 | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/12/) | 2023-12-06 18:46:26 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231206-184428-4aa7f1-pr4382.aar) |
| :heavy_check_mark: | 4aa7f108 | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/12/) | 2023-12-06 18:47:43 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231206-184428-4aa7f1-pr4382.zip) |
| | | | | | | |
| :heavy_check_mark: | a1ff389e | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/13/) | 2023-12-06 18:49:39 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231206-184813-a1ff38-pr4382.zip) |
| :heavy_check_mark: | a1ff389e | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/13/) | 2023-12-06 18:49:54 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231206-184813-a1ff38-pr4382.aar) |
| :heavy_check_mark: | a1ff389e | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/13/) | 2023-12-06 18:50:50 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231206-184813-a1ff38-pr4382.zip) |
| :heavy_check_mark: | a1ff389e | [#22](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/22/) | 2023-12-06 19:21:26 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/22//consoleText) |
| | | | | | | |
| :heavy_check_mark: | cd37dccb | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/14/) | 2023-12-06 23:38:15 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231206-233650-cd37dc-pr4382.zip) |
| :heavy_check_mark: | cd37dccb | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/14/) | 2023-12-06 23:38:27 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231206-233650-cd37dc-pr4382.aar) |
| :heavy_check_mark: | cd37dccb | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/14/) | 2023-12-06 23:39:58 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231206-233650-cd37dc-pr4382.zip) |
| :heavy_check_mark: | cd37dccb | [#23](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/23/) | 2023-12-07 00:09:51 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/23//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 8787f9e4 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/15/) | 2023-12-06 23:41:04 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231206-233949-8787f9-pr4382.zip) |
| :heavy_check_mark: | 8787f9e4 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/15/) | 2023-12-06 23:41:26 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231206-233945-8787f9-pr4382.aar) |
| :heavy_check_mark: | 8787f9e4 | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/15/) | 2023-12-06 23:42:58 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231206-234001-8787f9-pr4382.zip) |
| :heavy_multiplication_x: | 8787f9e4 | [#24](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/24/) | 2023-12-07 00:27:35 | ~17 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/24//consoleText) |
| :heavy_multiplication_x: | 8787f9e4 | [#25](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/25/) | 2023-12-07 00:58:07 | ~8 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/25//consoleText) |
| :heavy_multiplication_x: | 8787f9e4 | [#26](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/26/) | 2023-12-07 02:11:11 | ~17 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/26//consoleText) |
| :heavy_check_mark: | 8787f9e4 | [#27](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/27/) | 2023-12-07 03:12:44 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/27//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 4f55d1f2 | [#28](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/28/) | 2023-12-08 05:43:16 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/28//consoleText) |
| :heavy_check_mark: | 4f55d1f2 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/16/) | 2023-12-08 05:43:50 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231208-054213-4f55d1-pr4382.zip) |
| :heavy_check_mark: | 4f55d1f2 | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/16/) | 2023-12-08 05:44:18 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231208-054209-4f55d1-pr4382.aar) |
| :heavy_check_mark: | 4f55d1f2 | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/16/) | 2023-12-08 05:45:23 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231208-054209-4f55d1-pr4382.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 48b04c5f | [#29](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/29/) | 2023-12-08 05:47:09 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/29//consoleText) |
| :heavy_check_mark: | 48b04c5f | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/17/) | 2023-12-08 05:47:34 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231208-054559-48b04c-pr4382.zip) |
| :heavy_check_mark: | 48b04c5f | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/17/) | 2023-12-08 05:48:01 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231208-054559-48b04c-pr4382.aar) |
| :heavy_check_mark: | 48b04c5f | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/17/) | 2023-12-08 05:48:43 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231208-054559-48b04c-pr4382.zip) |
| | | | | | | |
| :heavy_check_mark: | b8da316e | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/18/) | 2023-12-08 06:00:13 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231208-055839-b8da31-pr4382.aar) |
| :heavy_check_mark: | b8da316e | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/18/) | 2023-12-08 06:00:13 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231208-055839-b8da31-pr4382.zip) |
| :heavy_check_mark: | b8da316e | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/18/) | 2023-12-08 06:01:45 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231208-055839-b8da31-pr4382.zip) |
| :heavy_check_mark: | b8da316e | [#30](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/30/) | 2023-12-08 06:31:58 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/30//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 4fe18d9e | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/19/) | 2023-12-08 06:50:26 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231208-064859-4fe18d-pr4382.zip) |
| :heavy_check_mark: | 4fe18d9e | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/19/) | 2023-12-08 06:51:14 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231208-064859-4fe18d-pr4382.aar) |
| :heavy_check_mark: | 4fe18d9e | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/19/) | 2023-12-08 06:51:48 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231208-064859-4fe18d-pr4382.zip) |
| :heavy_check_mark: | 4fe18d9e | [#31](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/31/) | 2023-12-08 07:21:53 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/31//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 995ce07b | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4382/20/) | 2023-12-08 07:03:38 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231208-070203-995ce0-pr4382.zip) |
| :heavy_check_mark: | 995ce07b | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4382/20/) | 2023-12-08 07:03:57 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231208-070159-995ce0-pr4382.aar) |
| :heavy_check_mark: | 995ce07b | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4382/20/) | 2023-12-08 07:05:07 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231208-070203-995ce0-pr4382.zip) |
| :heavy_multiplication_x: | 995ce07b | [#32](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/32/) | 2023-12-08 07:38:20 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/32//consoleText) |
| :heavy_check_mark: | 995ce07b | [#33](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/33/) | 2023-12-08 08:24:57 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4382/33//consoleText) |

osmaczko reviewed (approved):
Nice, congrats for first status-go contribution :+1:

jrainville commented on protocol/message_persistence.go:
```diff
@@ -1891,6 +1891,58 @@ func (db sqlitePersistence) MarkMessagesSeen(chatID string, ids []string) (uint6
+		return 0, 0, err
+	}
+
+	_, err = tx.Exec(`UPDATE user_messages SET seen = 0 WHERE seen = 1 AND NOT(mentioned) AND NOT(replied) AND id = ?`, messageId)
```
I think the reason why the count of unread mentions doesn't update correctly is because you are only marking as unread the one message you selected.

I think you're only missing a SELET to get the timestamp of the message that was clicked, then instead of updating using the `id`, use a `> timestamp` to make sure all newer messages as **also** marked as unread!

kounkou replied:
> I think the reason why the count of unread mentions doesn't update correctly is because you are only marking as unread the one message you selected.
> 
> I think you're only missing a SELET to get the timestamp of the message that was clicked, then instead of updating using the `id`, use a `> timestamp` to make sure all newer messages as **also** marked as unread!

oh... that's interesting perspective. Thanks, let me have a look

jrainville reviewed (commented):
Great work! I added a comment inline that explains what might work to make the mentions count work correctly.

kounkou committed:
feature(@desktop/chat) Enhance message context menu with mark as unread

fixes #10329

linked with PR #12879

- Adds capacity to mark a message as unread
- Adds capacity to mark a message with mention as unread
- Adds persistence to the marking of the message (change can be seen at
  after reboot)
- Adds marking in right click contextual menu

kounkou commented:
Working on fixing the unittests. 

Edit : 
Seems unrelated failures

```
TestMessengerCommunitiesTokenPermissionsSuite – github.com/status-im/status-go/protocol
TestMessengerCommunitiesTokenPermissionsSuite/TestBecomeMemberPermissions – github.com/status-im/status-go/protocol
TestMessengerCommunitiesTokenPermissionsSuite/TestCommunityTokensMetadata – github.com/status-im/status-go/protocol
TestMessengerCommunitiesTokenPermissionsSuite/TestCreateTokenPermission – github.com/status-im/status-go/protocol
TestMessengerCommunitiesTokenPermissionsSuite/TestEditSharedAddresses – github.com/status-im/status-go/protocol
TestMessengerCommunitiesTokenPermissionsSuite/TestEditTokenPermission – github.com/status-im/status-go/protocol
```

Those tests are flaky. After multiple retries, the tests are Pass

igor-sirotin commented on protocol/messenger.go:
```diff
@@ -4138,6 +4138,10 @@ func (m *Messenger) DeleteMessagesByChatID(id string) error {
 }
 
+func (m *Messenger) MarkMessageAsUnread(chatID string, messageID string) (uint64, uint64, error) {
+	return m.persistence.MarkMessageAsUnread(chatID, messageID)
```
Should we update `m.allChats` here as done in `markMessagesSeenImpl`?
https://github.com/status-im/status-go/blob/6197ed1bdb4b14ec7454e34086bafb2cf0f4952c/protocol/messenger.go#L4139-L4143

kounkou replied:
I am not sure why we need this? Could you explain why this is needed please?

kounkou replied:
After second thought I understand why we need to store the the chat in the map. Everything is explained here : https://github.com/status-im/status-go/issues/2203
This is legacy code that we might want to work on to as it takes mental effort to understand why we should put this line in the first place.

igor-sirotin commented on protocol/message_persistence.go:
```diff
@@ -1891,6 +1891,74 @@ func (db sqlitePersistence) MarkMessagesSeen(chatID string, ids []string) (uint6
+		_ = tx.Rollback()
+	}()
+
+	_, err = tx.Exec(`UPDATE user_messages SET seen = 1 WHERE NOT(seen)`)
```
Looks like this line will mark ALL of the messages as `seen`.
Even the ones from different chats.

kounkou replied:
Thanks for the comment, I got some doubts, and as discussed offline, I went into testing this. As a prerequesite, I followed the notion document to be able to decrypt messages in database.
Here are the steps I went through for the testing:


I made a test account test-jacques have 2 contacts (Jacques and Tester1) to chat with.
I created 2 chats, to get messages from those 2 accounts
Then I sent some messages,
Here I ran a query after decrypting messages in database :
        
```bash
sqlite> select text, seen from user_messages;
You sent a contact request to @0x04eeb4914b7eb91000de56f00fcb78845d461d0cacd84ba6cb2ba1c149665326fafcfb4c00cb7fa88c3c95b245eaaf8ae1a98bb89c67d98ef46a1c70c7bb4c131e|1
Hi !|1
Tester1 : in here|1
Tester1 : ahah|1
test-jacques : being the boss|0
test-jacques : I am the one|0
Tester1 : hello|1
Tester1 : I|1
test-jacques : hello|0
test-jacques : yo!|0
test-jacques : ahah|0
Tester1 : am|1
Tester1 : the boss|1

```

then from test-jacques, I went into one of the chats with Test1 and moved the marker around,
the expectation is that all the messages will be marked as seen

```bash
sqlite> select text, seen from user_messages;
You sent a contact request to @0x04eeb4914b7eb91000de56f00fcb78845d461d0cacd84ba6cb2ba1c149665326fafcfb4c00cb7fa88c3c95b245eaaf8ae1a98bb89c67d98ef46a1c70c7bb4c131e|1
Hi !|1
Tester1 : in here|1
Tester1 : ahah|0
test-jacques : being the boss|0
test-jacques : I am the one|0
Tester1 : hello|1
Tester1 : I|1
test-jacques : hello|0
test-jacques : yo!|0
test-jacques : ahah|0
Tester1 : am|1
Tester1 : the boss|1
```

Conclusion : 

We can see we do not impact the other chat here for which the messages start with `test-jacques`
So why is it the case ? 
The reason is that at the end of the my function I do the following which restrict my query to the provided chatId.
I hope this answers your concern

```sql
	// Update denormalized count
	_, err = tx.Exec(
		`UPDATE chats
              	SET unviewed_message_count =
		   (SELECT COUNT(1)
		   FROM user_messages
		   WHERE local_chat_id = ? AND seen = 0),
		   unviewed_mentions_count =
		   (SELECT COUNT(1)
		   FROM user_messages
		   WHERE local_chat_id = ? AND seen = 0 AND (mentioned OR replied)),
                   highlight = 0
		WHERE id = ?`, chatID, chatID, chatID)
	return countWithMentions + countNoMentions, countWithMentions, err
```

osmaczko replied:
> The reason is that at the end of the my function I do the following which restrict my query to the provided chatId.

Isn't it the restriction for different query? :thinking:

kounkou replied:
I will add another layer of protection by using the chatId.
But let me know if you see any flaws in my testing please
I will allocate time for myself to understand why the experiment doesn't show any impact at all

kounkou replied:
I have created a test :
see : 
`s.T().Run("Test marking a message as unread in a chat does not impact other chat", func(t *testing.T) {` 

Which helps me show that if we do not make the queries specific to a chatID, we can impact other chats as the messages inside the user_messages table contains messages from different chats.
I have also launched faulty commands 1 by 1, and could see how not providing the chatID impacts other chats.
So I believe this topic can be closed.

igor-sirotin commented on protocol/message_persistence.go:
```diff
@@ -1891,6 +1891,74 @@ func (db sqlitePersistence) MarkMessagesSeen(chatID string, ids []string) (uint6
+		`UPDATE user_messages 
+			SET seen = 0 
+			WHERE seen = 1 
+			AND (mentioned OR replied) 
```
It took me awhile to get why we do 2 separate requests for `mentioned or replied` messages.
Maybe instead we could do it in a more straightforward way?
1. `UPDATE ... SET seen = 1 WHERE timestamp > ...`
2. `SELECT COUNT(1) FROM user_messages WHERE seen = 0`
3. `SELECT COUNT(1) FROM user_messages WHERE seen = 1`

IMO the intentions would be more obvious like thie.
Also, if I get it right, we will cut the number of db requests from 4 to 3 (if `SELECT changes()` are counted).

kounkou replied:
Ah, I was following the same pattern as other code I learnt from, but let me test having a single query here 👍🏾

kounkou replied:
I tried as my best on this one, but I couldn't make the provided idea work. Let's discuss offline if needed, for now I have added a TODO note for this item, and submitted changes for the other comments to make progress.

igor-sirotin replied:
I wonder why this didn't work. Let's discuss in DM.

igor-sirotin reviewed (changes requested):
Good job, congrats on first status-go PR! 🎉 
Left some comments, some of them look very important.

Also, do you think we could implement a test for this feature?
Looks like not too complicated one, should be a good option for a first status-go test as well 🙂

kounkou commented:
> Good job, congrats on first status-go PR! 🎉 Left some comments, some of them look very important.
> 
> Also, do you think we could implement a test for this feature? Looks like not too complicated one, should be a good option for a first status-go test as well 🙂

Implemented unit-tests in the latest revision, and added some comments to get everybody up to speed for understanding what's happening. cheers!

igor-sirotin commented on protocol/messenger_test.go:
```diff
@@ -2540,3 +2540,237 @@ func (s *MessengerSuite) TestSendMessageMention() {
+
+		chats := s.m.Chats()
+
+		for idx := range chats {
```
This function can be used to avoid using a loop:
https://github.com/status-im/status-go/blob/3ab1afaae8c01c3228a8349fb606b767e7d697dc/protocol/messenger_chats.go#L135-L139

kounkou replied:
sure, I can update this. Thanks

igor-sirotin commented on protocol/messenger_test.go:
```diff
@@ -2540,3 +2540,237 @@ func (s *MessengerSuite) TestSendMessageMention() {
+		}
+
+		count, countWithMentions, err := s.m.MarkMessageAsUnread(chat.ID, inputMessage1.ID)
+		s.Require().NoError(err)
+		s.Require().Equal(uint64(1), count)
+		s.Require().Equal(uint64(1), countWithMentions)
+
+		chats = s.m.Chats()
+
+		for idx := range chats {
+			if chats[idx].ID == chat.ID {
+				s.Require().Equal(uint(1), chats[idx].UnviewedMessagesCount)
+				s.Require().Equal(uint(1), chats[idx].UnviewedMentionsCount)
+				s.Require().Equal(false, chats[idx].Highlight)
+			}
+		}
```
Here you check 2 things:
- return value of `MarkMessageAsUnread`
- chat's `UnviewedMessagesCount` and `UnviewedMentionsCount`

It's good, but I think the most important part is to check that the messages `Seen` value was indeed updated. And that it's correct for each message. You can use `MessageByID` function for this. I think you have IDs of all saved messages.

kounkou replied:
Fair enough, updated the test. Thanks

igor-sirotin commented on protocol/messenger_test.go:
```diff
@@ -2540,3 +2540,237 @@ func (s *MessengerSuite) TestSendMessageMention() {
+		inputMessage1.Seen = false
+		inputMessage1.Mentioned = false
+
+		time.Sleep(1000 * time.Millisecond)
```
a leftover or a hack?

kounkou replied:
was a leftover, removed. Thanks

igor-sirotin commented on protocol/messenger_test.go:
```diff
@@ -2540,3 +2540,237 @@ func (s *MessengerSuite) TestSendMessageMention() {
+		count, countWithMentions, err := s.m.MarkMessageAsUnread(chat1.ID, inputMessage1.ID)
+
+		s.Require().NoError(err)
+		s.Require().Equal(uint(1), chat2.UnviewedMessagesCount)
```
hmm, I don't think it makes sense to check this, would this value update automatically?
Even though `chat2` is a pointer, we replace it in `allChats` with a new Chat later.

kounkou replied:
This test is useful for detecting regression. From my tests, if I remove the `WHERE local_chat_id = ?` from the query, I could see impact on the second chat.
I have improved the test also based on previous comments to test the .Seen property

igor-sirotin commented on protocol/messenger_test.go:
```diff
@@ -2540,3 +2540,237 @@ func (s *MessengerSuite) TestSendMessageMention() {
+				s.Require().Equal(uint(2), chats[idx].UnviewedMessagesCount)
+				s.Require().Equal(uint(0), chats[idx].UnviewedMentionsCount)
+				s.Require().Equal(false, chats[idx].Highlight)
+			} else if chats[idx].ID == chat2.ID {
```
This is not enough here.
Since you don't update chat2 in `messenger.allChats`, the value will surely remain the same.
We should check that the `Seen` property for the message in the database in correct.

kounkou replied:
that's right, updated all tests to account for the same, meaning testing the .Seen property.

igor-sirotin commented on protocol/messenger_test.go:
```diff
@@ -2540,3 +2540,237 @@ func (s *MessengerSuite) TestSendMessageMention() {
 	s.Require().Equal("Alice talk to bobby", response.Notifications()[0].Message)
 }
+
+func (s *MessengerSuite) TestMarkMessageAsUnread() {
```
Overall the test look very good, thank you! I hope you enjoyed this 🙂 

I would recommend moving this to a separate `messenger_unread_test.go`, 2500 lines of code is more than enough here already 😄  And if you create a separate `MessengerUnreadSuite`, you can implement each test in a separate function and you won't have to call `s.T().Run(...)`.

igor-sirotin commented on protocol/message_persistence.go:
```diff
@@ -1891,6 +1891,78 @@ func (db sqlitePersistence) MarkMessagesSeen(chatID string, ids []string) (uint6
+	_, err = tx.Exec(
+		`UPDATE chats
+              	SET unviewed_message_count =
+		   (SELECT COUNT(1)
+		   FROM user_messages
+		   WHERE local_chat_id = ? AND seen = 0),
```
hmm, but we have the `countNoMentions` and `countWithMentions` already, maybe just push it here as a parameter? Should be better on performance I guess?

kounkou replied:
Fair enough, updated the code

igor-sirotin reviewed (commented):
I dont' have any concerns on the implementation now, but have some important comments on the test. Not a blocker, but please check out 🙂 

Thanks for writing a test! ❤️

igor-sirotin commented on protocol/messenger_unread_test.go:
```diff
@@ -0,0 +1,277 @@
+
+	actualChat, ok := chats.Load(chat.ID)
+
+	s.Require().Equal(true, ok)
```
```suggestion
	s.Require().True(ok)
```

igor-sirotin commented on protocol/messenger_unread_test.go:
```diff
@@ -0,0 +1,277 @@
+	s.Require().Equal(true, ok)
+	s.Require().Equal(uint(0), actualChat.UnviewedMessagesCount)
+	s.Require().Equal(uint(0), actualChat.UnviewedMentionsCount)
+	s.Require().Equal(false, actualChat.Highlight)
```
```suggestion
	s.Require().False(actualChat.Highlight)
```

igor-sirotin replied:
this and all similar places below

igor-sirotin commented on protocol/messenger_unread_test.go:
```diff
@@ -0,0 +1,277 @@
+	inputMessage1.Seen = true
+	inputMessage1.Mentioned = true
+
+	time.Sleep(1000 * time.Millisecond)
```
still leftover? 🙂

kounkou replied:
I used it to simulate the timestamp as the timestamp is created automatically when creating the message and to account for the speed of unittest, but I understand I can directly overid the Timestamp to make the test more controllable here. I'll update this. thanks

igor-sirotin commented on protocol/messenger_unread_test.go:
```diff
@@ -0,0 +1,277 @@
+	s.Require().Equal(uint(0), actualChat.UnviewedMentionsCount)
+	s.Require().Equal(false, actualChat.Highlight)
+
+	actualMessage1, err := s.m.MessageByID(inputMessage1.ID)
+	s.Require().NoError(err)
+	s.Require().Equal(true, actualMessage1.Seen)
```
So you could make a separate function and simplify the code a lot.

```go
func (s *MessengerSuite) checkMessageSeen(messageID string, expectedSeen) {
    message, err := s.m.MessageByID(messageID)
	s.Require().NoError(err)
	s.Require().Equal(expectedSeen, message.Seen)
}

s.checkMessageSeen(inputMessage1.ID, true)
s.checkMessageSeen(inputMessage2.ID, true)
s.checkMessageSeen(inputMessage3.ID, true)
```

kounkou replied:
Sure, updated. thanks

igor-sirotin commented on protocol/messenger_unread_test.go:
```diff
@@ -0,0 +1,277 @@
+
+	inputMessage3 := buildTestMessage(*chat2)
+	inputMessage3.ID = "3"
+	inputMessage3.Seen = false
```
hmmm, but it's `false` from the start of the test 🤔  Shouldn't it be `true` here?
And same for `inputMessage1` and `inputMessage2`.

kounkou replied:
This is chat2, so it's never touched 😄 so, it should be false, not true. Nonetheless I have made it obvious now that for chat1 the seen states are changing as improvement.

MishkaRogachev commented on protocol/message_persistence.go:
```diff
@@ -1891,6 +1891,73 @@ func (db sqlitePersistence) MarkMessagesSeen(chatID string, ids []string) (uint6
 	return countWithMentions + countNoMentions, countWithMentions, err
 }
 
+func (db sqlitePersistence) MarkMessageAsUnread(chatID string, messageID string) (uint64, uint64, error) {
```
just suggestion: you can parametrise `MarkMessagesSeen` with boolean seen/unseen flag instead of this function

kounkou replied:
MarkAsUnread is more than just making a bunch of items as unread. The difference is that with MarkAsUnread, we also need to mark every items with smaller timestamp than the current element as read. The name mark as read is still valid though. So that reason I think a separate function makes more sense.

MishkaRogachev commented on protocol/messenger.go:
```diff
@@ -4138,6 +4138,31 @@ func (m *Messenger) DeleteMessagesByChatID(id string) error {
+	if err != nil {
+		return 0, 0, err
+	}
+
```
i think it's right to also check activity center here. If persistence has AC notification with `id == messageID` it also should be marked as unseen. Basically same as here: https://github.com/status-im/status-go/blob/develop/protocol/messenger.go#L4173

kounkou replied:
> i think it's right to also check activity center here. If persistence has AC notification with `id == messageID` it also should be marked as unseen. Basically same as here: https://github.com/status-im/status-go/blob/develop/protocol/messenger.go#L4173

Let me have a look.

MishkaRogachev replied:
Agh i probably got the idea, `persistence.MarkMessageAsUnread` function marks as unread not the message with provided id, but all seen in the chat starting with a timestamp of the message with provided id. Name is quite confusing..

Wouldn't it be easier with this approach:
1) Rename the function to `MarkMessagesAsUnreadSinceTimestamp`, and pass the desired timestamp to the status-go instead of the message id
2) Call `persistence.MarkMessagesAsUnreadSinceTimestamp`, which will update the `user_messages` and `chats` tables and return either a slice of messages or a slice of message id, offected by updates.
3) Call `persistence.MarkActivityCenterNotificationsUnread`, with the message ids we got in the previous call

kounkou replied:
> Agh i probably got the idea, `persistence.MarkMessageAsUnread` function marks as unread not the message with provided id, but all seen in the chat starting with a timestamp of the message with provided id. Name is quite confusing..
> 
> Wouldn't it be easier with this approach:
> 
> 1. Rename the function to `MarkMessagesAsUnreadSinceTimestamp`, and pass the desired timestamp to the status-go instead of the message id
> 2. Call `persistence.MarkMessagesAsUnreadSinceTimestamp`, which will update the `user_messages` and `chats` tables and return either a slice of messages or a slice of message id, offected by updates.
> 3. Call `persistence.MarkActivityCenterNotificationsUnread`, with the message ids we got in the previous call

1.
It's better to pass the messageId because passing the timestamp can be flawed. The source of truth for the timestamps is in the table. That's why I pass the messageId

Thanks for the suggestions

2.
For performance, it's better to keep those 2 separate because getting the messageIds or the Ids it requires to go over the results. To keep the `MarkMessageAsUnread` simple with regards to performance, it's better to not include the getTimestamps.

3. This is done already

igor-sirotin reviewed (approved):
🎉

MishkaRogachev reviewed (approved):
LGTM!
